lib/ansible/modules/storage/dellemc/dellemc_pmax_createsg.py

```diff
@@ -0,0 +1,200 @@
+#!/usr/bin/python
+# Copyright (C) 2018 DellEMC
+# Author(s): Paul Martin <paule.martin@dell.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+ANSIBLE_METADATA = {
+    'metadata_version': '1.1',
+    'status': ['preview'],
+    'supported_by': 'community'
+}
+
+DOCUMENTATION = '''
+---
+author:
+  - "Paul Martin (@rawstorage)"
+short_description: "Create storage group on Dell EMC PowerMax or VMAX All
+Flash"
+version_added: "2.8"
+description:
+  - "This module has been tested against UNI 9.0. Every effort has been made
+  to verify the scripts run with valid input. These modules are a tech preview"
+module: dellemc_pmax_createsg
+options:
+  array_id:
+    description:
+      - "Integer 12 Digit Serial Number of PowerMAX or VMAX array."
+    required: true
+  cap_unit:
+    choices:
+      - GB
+      - TB
+      - MB
+      - CYL
+    description:
+      - "String value, default is set to GB"
+    required: false
+  num_vols:
+    description:
+      - "integer value for the number of volumes. Minimum is 1, module will
+      fail if less than one volume is specified or value is 0. If volumes are
+      required of different sizes, additional tasks should be added to
+      playbooks to use dellemc_pmax_addvolume module"
+    required: true
+  sgname:
+    description:
+      - "Storage Group name 32 Characters no special characters other than
+      underscore"
+    required: true
+  slo:
+    description:
+      - "Service Level for the storage group, Supported on VMAX3 and All Flash
+      and PowerMAX NVMe Arrays running PowerMAX OS 5978 and above.  Default is
+      set to Diamond, but user can override this by setting a different value."
+    required: false
+  srp_id:
+    description:
+      - "Storage Resource Pool Name, Default is set to SRP_1, if your system
+      has mainframe or multiple pools you can set this to a different value to
+      match your environment"
+    required: false
```

Comment:
Specify type for all parameters like 

```suggestion
    required: false
    type: str
```